Source/WebCore/dom/ScriptElement.h

```diff
@@ -105,9 +105,13 @@ class ScriptElement {
 
     void didFinishInsertingNode();
     void childrenChanged(const ContainerNode::ChildChange&);
+    void finishParsingChildren();
     void handleSourceAttribute(const String& sourceURL);
     void handleAsyncAttribute();
 
+    // https://w3c.github.io/trusted-types/dist/spec/#slots-with-trusted-values
+    String m_scriptText { emptyString() };
```

Comment:
Why is this public? It should be protected at worst.

Reply:
Ah apologies that's an oversight on my part it should be protected as you say

Reply:
Any reason to not let this be null string instead of empty string?

Reply:
@rniwa, it is protected.

But since the derived class only needs to **set** `m_scriptText`, I think there should be a protected **function** for it to call, like maybe named something like `setTrustedScriptText` and `m_scriptText` itself can be private. I also think that `m_scriptText` should be named `m_trustedScriptText` for greater clarity.

Reply:
> Any reason to not let this be null string instead of empty string?

Yes there is, I've just realised there isn't a web platform test that covers the difference so I will add one.

Effectively if it's a null string then an empty script created through DOM APIs would lead to a TT violation (because the script content is never a null string).

Reply:
Turns out the existing tests do cover this just not super explicitly. It's also quite hard to test explicitly because you're testing for the lack of an event.